ibis/backends/pandas/client.py

```diff
@@ -301,23 +301,23 @@ def convert_array_to_series(in_dtype, out_dtype, column):
 sch.Schema.to_pandas = ibis_schema_to_pandas  # type: ignore
 
 
-class DataFrameProxy(Immutable):
+class DataFrameProxy(Immutable, util.ToFrame):
     __slots__ = ('_df', '_hash')
 
     def __init__(self, df):
         object.__setattr__(self, "_df", df)
         object.__setattr__(self, "_hash", hash((type(df), id(df))))
 
-    def __getattr__(self, name):
-        return getattr(self._df, name)
-
     def __hash__(self):
         return self._hash
 
     def __repr__(self):
         df_repr = util.indent(repr(self._df), spaces=2)
         return f"{self.__class__.__name__}:\n{df_repr}"
 
+    def to_frame(self):
```

Comment:
I'm not familiar with `DataFrameProxy` objects - are these user facing? If so:

- Perhaps this should be `to_pandas(self)`, mirroring conventions used in other projects (e.g. `pyarrow.Table.to_pandas`)
- Perhaps this should have a docstring

Reply:
`DataFrameProxy` is not user facing. I personally prefer `to_frame()` since this is never dealing with Series. I'd like to be able to read the code and know that I'm getting a DataFrame, rather than just any ol' pandas object.